workflow/executor.go

```diff
@@ -22,17 +22,24 @@ func NewExecutor(
 	logger log.Logger,
 	config *config.Config,
 	stepRegistry step.Registry,
+	callableFunctions map[string]schema.CallableFunction,
 ) (Executor, error) {
 	if logger == nil {
 		return nil, fmt.Errorf("bug: no logger passed to NewExecutor")
 	}
 	if stepRegistry == nil {
 		return nil, fmt.Errorf("bug: no step registry passed to NewExecutor")
 	}
+	functionSchemas := make(map[string]schema.Function, len(callableFunctions))
+	for k, v := range callableFunctions {
+		functionSchemas[k] = v
+	}
```

Comment:
I feel like the type choices aren't serving us well if we have to play games like this.  (Making a copy of the map just to change the value type, when the values are assignable and therefore castable, is just a waste of time and space.)

It looks to me like you could cast the resulting value (which would be a `schema.CallableFunction`), at the point where the map is referenced, into a `schema.Function`...isn't that true?

Anywhere where you have a `schema.CallableFunction`, you have access to the embedded `schema.Function` interface, so there shouldn't be a functional problem...the question is what is required to satisfy Go's type requirements...can't we do that without duplicating the map?

Reply:
So, then, why is the "given interface" appropriate?  Why maintain separate types for `schema.CallableFunction` and  `schema.Function`?  Or, why not use the `schema.Function` type for the list?  (If the values are actually `schema.CallableFunction` instances, then the receiver can up-cast them if it needs that interface.)

Reply:
Requiring the interface use CallableFunction is limiting, because CallableFunction is the least-portable form of it. It cannot be serialized and unserialized in a way that preserves the handler. So while with the existing design, if you don't have the function handlers, you can at least resolve dependencies. Making this change would make that not possible.
Yes, you could argue that the trade-off is worth it. But it's not much overhead as written.

Reply:
So yes, `schema.CallableFunction` provides all of the context. However, it's not portable. It made sense for the API call to use the more portable interface when the specific handlers weren't necessary.